android-components/components/lib/state/src/main/java/mozilla/components/lib/state/ext/ComposeExtensions.kt

```diff
@@ -43,6 +44,35 @@ fun <S : State, A : Action, R> Store<S, A>.observeAsComposableState(map: (S) ->
     return state
 }
 
+/**
+ * Starts observing this [Store] and represents the mapped state (using [map]) via [ComposeState].
+ *
+ * Every time the mapped [Store] state changes, the returned [ComposeState] will be updated causing
+ * recomposition of every [ComposeState.value] usage.
+ *
+ * The [Store] observer will automatically be removed when this composable disposes or the current
+ * [LifecycleOwner] moves to the [Lifecycle.State.DESTROYED] state.
+ *
+ * @param initialMappedValue Initial value emitted.
+ * @param map The applied function to produced the mapped value [R] from [S].
+ * @return A non nullable [ComposeState], making the api more reasonable for callers where the
+ * state is non null.
+ */
+@Composable
+fun <S : State, A : Action, R> Store<S, A>.observeAsComposableState(
```

Comment:
I am wondering if there's a reason we would want to keep the above `observeAsComposableState` that would return the nullable `ComposeState` after this?

Reply:
Yeah it's worth exploring whether we can pass in `store.state.r` as `initialValue` for all cases and then maybe remove the other variant.

Reply:
I filed [a bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1832624) to track this as a follow-up, because I think it's worth doing.

Reply:
Thanks for creating that bug!